### PR TITLE
Update base model edit destroy

### PR DESCRIPTION
**Pull request by allouis**
No ticket - makes the edit and destroy models suitable for overriding in a subclass

Had to update other models for not correctly calling their super methods

### CODE REVIEW

File: core/server/models/base/index.js [hunk 1]
```diff
@@ -498,12 +498,18 @@ ghostBookshelf.Model = ghostBookshelf.Model.extend({
...
-            return ['shallow', 'withRelated', 'context', 'columns', 'absolute_urls'];
+        const baseOptions = ['context', 'withRelated'];
+        const extraOptions = ['transacting', 'importing', 'forUpdate', 'migrating'];
+
+        switch (methodName) {
+        case 'toJSON':
+            return baseOptions.concat('shallow', 'columns', 'absolute_urls');
+        case 'destroy':
+        case 'edit':
+            return baseOptions.concat(extraOptions, ['id']);
```
Comment (kirrg001) [inline core/server/models/base/index.js]: e.g. the user model re-defines `id`, [see](https://github.com/TryGhost/Ghost/blob/2.1.3/core/server/models/user.js#L338).

File: core/server/models/app.js [hunk 2]
```diff
@@ -34,7 +34,7 @@ App = ghostBookshelf.Model.extend({
     * @return {Array} Keys allowed in the `options` hash of the model's method.
     */
     permittedOptions: function permittedOptions(methodName) {
-        var options = ghostBookshelf.Model.permittedOptions(),
+        var options = ghostBookshelf.Model.permittedOptions.call(this, methodName),
```
Comment (kirrg001) [inline core/server/models/app.js]: General: There are some more models to update, or? 🤔 
Why did you only update these three?

File: core/server/models/base/index.js (hunk 1 above)
Comment (allouis) [reply core/server/models/base/index.js]: it's also been re-defining `withRelated` too - should that be done in a separate cleanup PR?

File: core/server/models/app.js (hunk 2 above)
Comment (allouis) [reply core/server/models/app.js]: missed the others - will update :relaxed:

File: core/server/models/base/index.js (hunk 1 above)
Comment (kirrg001) [reply core/server/models/base/index.js]: Don't have to as part of this PR. But you have added the `id` property as default for edit/destroy, i would only look for this?

File: core/server/models/app.js (hunk 2 above)
Comment (allouis) [reply core/server/models/app.js]: updated btw!

Comment (kirrg001) [reply core/server/models/app.js]: 👍

File: core/server/models/base/index.js (hunk 1 above)
Comment (kirrg001) [reply core/server/models/base/index.js]: User model was updated

**Review by kirrg001 — APPROVED** [review]

**Comment by allouis** [comment]
Fixed commit messages :+1: There should only be one build for this - github sent a `push` request to travis, and travis can't find the branch. That's why the build is failing for `push`

**Pull request merged**